### DISCUSSION TITLE
Руководство ПЛИ, тело документа, п.1.7 Секция "Связанные документы" LINKDOCS

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#286] [Лабораторная РГ]
Секция предназначена для указания сведений о связанных документах. В данную секцию включается следующая информация:
• Вид документа;
• Примечание к документу (опционально);
• Дата формирования документа;
• Серия документа (опционально);
• Номер документа (опционально);
• Форма документа (опционально);
• Уникальный идентификатор документа в МИС (при наличии);
• Уникальный идентификатор документа в РЭМД (при наличии).

**Comment by eugenekogan-spb** [comment]
К сожалению, в качестве примера приведена пустая секция
<img width="751" alt="image" src="https://github.com/user-attachments/assets/1fd139b6-fdef-4ce1-9406-5f55bd825209">

**Comment by eugenekogan-spb** [comment]
У3-6: Секция "Связанные документы" ОБЯЗАНА иметь хотя бы один [1..*] элемент
entry, содержащий сведения о связанном документе, который:
1. ОБЯЗАН иметь один [1..1] элемент act (атрибут @classCode должен иметь
значение "ACT", атрибут @moodCode должен иметь значение "EVN"), который:
1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1522", атрибут @codeSystemName должен иметь значение "Виды медицинской документации", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий вид документа;
2.МОЖЕТ иметь один [0..1] элемент text, содержащий примечание к документу, который должен иметь не пустое наполнение;
3. ОБЯЗАН иметь один [1..1] элемент effectiveTime (атрибут @value должен быть представлен с не пустым значением), содержащий дату формирования документа;
4. МОЖЕТ иметь один [0..1] элемент entryRelationship (атрибут @typeCode должен иметь значение "COMP"), содержащий серию документа, который:
1.ОБЯЗАН иметь один [1..1] элемент observation (атрибут @classCode должен иметь значение "OBS", атрибут @moodCode должен иметь значение "EVN"), который:
1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @code должен иметь значение "11002", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.166", атрибут @codeSystemName должен иметь значение "Кодируемые поля CDA документов", атрибут @displayName должен иметь значение "Серия документа", атрибут @codeSystemVersion должен быть представлен с не пустым значением), содержащий код поля;
2. ОБЯЗАН иметь один [1..1] элемент value (атрибут @xsi:type должен иметь значение "ST"), содержащий серию документа, который должен иметь не пустое наполнение;
5. МОЖЕТ иметь один [0..1] элемент entryRelationship (атрибут @typeCode должен иметь значение "COMP"), содержащий номер документа, который:
1.ОБЯЗАН иметь один [1..1] элемент observation (атрибут @classCode должен иметь значение "OBS", атрибут @moodCode должен иметь значение "EVN"), который:
Руководство по реализации СЭМД "Протокол лабораторного исследования" в соответствии с третьим уровнем требований архитектуры клинических документов версии 2 (CDA 2.0 HL7 v.3)
 Страница 111
1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @code должен иметь значение "11003", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.166", атрибут @codeSystemName должен иметь значение "Кодируемые поля CDA документов", атрибут @displayName должен иметь значение "Номер документа", атрибут @codeSystemVersion должен быть представлен с не пустым значением), содержащий код поля;
2. ОБЯЗАН иметь один [1..1] элемент value (атрибут @xsi:type должен иметь значение "ST"), содержащий номер документа, который должен иметь не пустое наполнение;
6. МОЖЕТ иметь один [0..1] элемент entryRelationship (атрибут @typeCode должен иметь значение "COMP"), содержащий форму документа, который:
1.ОБЯЗАН иметь один [1..1] элемент observation (атрибут @classCode должен иметь значение "OBS", атрибут @moodCode должен иметь значение "EVN"), который:
1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @code должен иметь значение "6058", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.166", атрибут @codeSystemName должен иметь значение "Кодируемые поля CDA документов", атрибут @displayName должен иметь значение "Форма документа", атрибут @codeSystemVersion должен быть представлен с не пустым значением), содержащий код поля;
2. ОБЯЗАН иметь один [1..1] элемент value (атрибут @xsi:type должен иметь значение "CD", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.1008", атрибут @codeSystemName должен иметь значение "Формы документов", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий форму документа;
7. ОБЯЗАН иметь один [1..1] элемент reference (атрибут @typeCode должен иметь значение "REFR"), содержащий ссылку на внешний документ, который:
1. ОБЯЗАН иметь один [1..1] элемент externalDocument (атрибут @classCode должен иметь значение "DOCCLIN", атрибут @moodCode должен иметь значение "EVN"), который:
1. ДОЛЖЕН иметь один [1..1] элемент id (атрибуты @root, @extension должны быть представлены с не пустыми значениями; атрибут @root должен быть сформирован по правилу формирования идентификаторов документов, т.е. "OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.51", атрибут @extension должен соответствовать идентификатору документа), содержащий уникальный идентификатор документа в МИС;
2. ДОЛЖЕН иметь один [1..1] элемент id (атрибут @root должен иметь значение "1.2.643.5.1.13.13.17.1.1", атрибут @extension должен соответствовать идентификатору документа в РЭМД (должен соответствовать регулярному выражению "^[0-9]{2}[.][0-9]{2}[.][0- 9]{3-4}[.][0-9]{9}$")), содержащий уникальный идентификатор документа в РЭМД.

**Comment by eugenekogan-spb** [comment]
Для передачи информации о связанных документах предусмотрен атрибут Composition.relatesTo, представляющий собой ссылку на ресурс RelatedArtifact
В R4 этот атрибут имеет код и ссылку на связанный документ Composition

Согласно требованиям данного раздела спецификации, в описании связанного документа необходимо предусмотреть:
- вид документации по справочнику "Виды медицинской документации" 1.2.643.5.1.13.13.11.1522
- дату формирования документа
- текстовый комментарий
- серию документа
- номер документа
- форма документа по справочнику "Формы документов" 1.2.643.5.1.13.13.99.2.1008
- уникальный идентификатор документа в МИС, пространство OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.51
- идентификатор документа в РЭМД, удовлетворяющий правилу "^[0-9]{2}[.][0-9]{2}[.][0- 9]{3-4}[.][0-9]{9}$"

Для R4
```
* relatesTo
  * code = #appends
  *  targetRefernce = Reference (composition-discharge-note)

Instance: composition-discharge-note
InstanceOf: Composition

* status = #final

* date = "2022-11-15T08:50:00+04:00"

* identifier
  * type
    * text = "Уникальный идентификатор документа в МИС"
  * system = "OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.51"
  * value = "1234"

* type
  * text = "community healthcare discharge summary"
  * coding
    * system = "loinc.org"
    * code = #100721-0

* author = Reference (organization-mariinskaya)

//только для R5
//* identifier[+]
//  * type
//    * text = "Уникальный идентификатор документа в РЭМД"
//  * system = "РЭМД"
//  * value = "какое-то-значение-в-стиле-^[0-9]{2}[.][0-9]{2}[.][0- 9]{3-4}[.][0-9]{9}$"

* title = "выписка из Мариинской больницы"

* category[+]
  * text = "Эпикриз в стационаре выписной"
  * coding
    * system = "1.2.643.5.1.13.13.11.1522"
    * code = #1

* category[+]
  * text = "Электронная"
  * coding
    * system = "1.2.643.5.1.13.13.99.2.1008"
    * code = #1

```
// Для R5 дополнительно
// Instance: related-artifact-discharge-note
// InstanceOf RelatedArtifact
// * type = #supported-with

Вопрос: как правильно передать серию и номер документа? Это не идентификатор, потому что не обязательно уникален.

**Reply by maximalexeev** [reply]
как передать серию и номер документа - а почему не через идентификатор в виде ""серия:номер"?

**Reply by eugenekogan-spb** [reply]
Потому что нужна уникальность. Нужно выдумать уникальный system, по аналогии с другими номерами, выданными внутри МО.

**Reply by eugenekogan-spb** [reply]
Пусть мы ссылаемся выписку из истории болезни №0004 за 2024 год

```
Instance: composition-discharge-note
InstanceOf: Composition

* identifier[+]
  * type
    * text = "Номер документа в медицинской организации"
  * system = "OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.1001"
  * value = "2024:0004"
```
